recipes/gnuradio/meta.yaml

```diff
@@ -0,0 +1,275 @@
+        - python
+        - six
+    test:
+      commands:
+        - gnuradio-config-info -v --prefix --sysconfdir --prefsdir --userprefsdir --prefs --builddate --enabled-components --cc --cxx --cflags
+        # run a few example scripts to show that at least some things work
+        - python "$CONDA_PREFIX/share/gnuradio/examples/volk_benchmark/volk_math.py" -L test -N 1e6 -I 3 --all  # [not win]
+        - python "%CONDA_PREFIX%/Library/share/gnuradio/examples/volk_benchmark/volk_math.py" -L test -N 1e6 -I 3 --all  # [win]
+      imports:
+        - gnuradio.analog
+        - gnuradio.audio
+        - gnuradio.blocks
+        - gnuradio.channels
+        - gnuradio.digital
+        - gnuradio.dtv
+        - gnuradio.fec
+        - gnuradio.fft
+        - gnuradio.filter
+        - gnuradio.gr
+        - gnuradio.trellis
+        - gnuradio.vocoder
+        - gnuradio.wavelet
+        - pmt
+    about:
+      summary: 'GNU Radio core functionality and modules'
+  - name: gnuradio-qtgui
+    script: install_qtgui.sh  # [not win]
+    script: install_qtgui.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - pkg-config  # [not win]
+      host:
+        - gnuradio-core
+        - boost-cpp
+        - log4cpp
+        - pyqt
+        - python
+        - qt
+        - qwt
+        - volk
+      run:
+        - {{ pin_subpackage('gnuradio-core', min_pin='x.x.x', max_pin='x.x.x') }}
+        - boost-cpp
+        - {{ pin_compatible('log4cpp') }}
+        - pyqt
+        - python
+        - {{ pin_compatible('qwt', max_pin='x.x') }}
+    test:
+      imports:
+        - gnuradio.qtgui
+    about:
+      summary: 'GNU Radio QT module providing graphical components'
+  - name: gnuradio-uhd
+    script: install_uhd.sh  # [not win]
+    script: install_uhd.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - pkg-config  # [not win]
+      host:
+        - gnuradio-core
+        - boost-cpp
+        - log4cpp
+        - python
+        - uhd
+      run:
+        - {{ pin_subpackage('gnuradio-core', min_pin='x.x.x', max_pin='x.x.x') }}
+        - boost-cpp
+        - {{ pin_compatible('log4cpp') }}
+        - python
+    test:
+      imports:
+        - gnuradio.uhd
+    about:
+      summary: 'GNU Radio UHD module for Ettus USRP radios'
+  - name: gnuradio-zeromq
+    script: install_zeromq.sh  # [not win]
+    script: install_zeromq.bat  # [win]
+    requirements:
+      build:
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - pkg-config  # [not win]
+      host:
+        - gnuradio-core
+        - boost-cpp
+        - cppzmq
+        - log4cpp
+        - python
+        - zeromq
+      run:
+        - {{ pin_subpackage('gnuradio-core', min_pin='x.x.x', max_pin='x.x.x') }}
+        - boost-cpp
```

Comment:
Is this needed at runtime? I think it only containers headers.
```suggestion
```

Reply:
No, `boost-cpp` also contains libraries but also has a `run_exports` so you can omit it as it will be automatically included.

Reply:
I'm testing this with a local build right now, but at some point in the past (could have been a while ago) I did find that the run dependency on ``boost-cpp`` was needed. I know it contains libraries that are needed at runtime, but @xhochy appears to be right that it should be automatic now. It's not a ``run_exports`` in the ``boost-cpp`` package currently, but it is set using ``pin_run_as_build`` in ``conda-forge-pinning``.

Reply:
My test without ``boost-cpp`` as a run requirement just failed locally with an overlinking error complaining about missing ``boost-cpp``. I guess ``pin_run_as_build`` doesn't add the dependency in the same way as ``run_exports``, it just sets the pin if it is there in the run requirements? Anyway, looks like these have to stay.

Reply:
How were you building this locally? The `pin_run_as_build` mechanism requires additional options to be passed (and might only work in feedstocks).

Reply:
Regardless, LGTM and it can be removed in the feedstock if needed 👍 Thanks @ryanvolz!

Reply:
I adapted ``build-locally.py`` from a feedstock run run ``.circleci/run_docker_build.sh`` in a docker container. Not sure if that would do it, but maybe?

Reply:
Anyway, I'll try it out with a pull request on the feedstock to make sure.

Reply:
> I adapted build-locally.py from a feedstock run run .circleci/run_docker_build.sh in a docker container. Not sure if that would do it, but maybe?

That will match with what is done for testing here in staged-recipes.

The `pin_run_as_build` constraints in the feedstocks come from rerendering where "variant config files" are generated and passed with something like `conda build -m .ci_support/config-something.yaml`.